learn/advanced/pagination.md

```diff
@@ -0,0 +1,226 @@
+
+  // Update page number
+  if (directionBtn === 'previous_button') {
+    pageNumber = pageNumber - 1
+  } else if (directionBtn === 'next_button') {
+    pageNumber = pageNumber + 1
+  }
+
+  // Store new page number in the pagination element
+  document.querySelector('.pagination').dataset.pageNumber = pageNumber
+}
+
+// Add data to our HTML element stating the user is on the first page
+document.querySelector('.pagination').dataset.pageNumber = 0
+// Each time a user clicks on the previous or next buttons, update the page number 
+document.querySelector('#previous_button').onclick = function () { updatePageNumber(this) }
+document.querySelector('#next_button').onclick = function () { updatePageNumber(this) }
+```
+
+#### Disabling navigation buttons for first and last pages
+
+It is often helpful to disable navigation buttons when the user cannot move to the "next" or "previous" page.
+
+The "previous" button should be disabled whenever your `offset` is 0, as this indicates your user is on the first results page.
+
+To know when to disable the "next" button, we recommend setting your query's `limit` to the number of results you wish to display per page plus one. That extra `hit` should not be shown to the user. Its purpose is to indicate that there is at least one more document to display on the next page.
+
+The following JavaScript snippet runs checks whether we should disable a button every time the user navigates to another search results page:
+
+```js
+function updatePageNumber() {
+  const pageNumber = parseInt(document.querySelector('.pagination').dataset.pageNumber)
+
+  const offset = pageNumber * 20
+  const results = await index.search('x', { limit: 21, offset })
+
+  // If offset equals 0, we're on the first results page
+  if (offset === 0 ) {
+    document.querySelector('#previous_button').disabled = true;
+  } 
+
+  // If offset is bigger than 0, we're not on the first results page
+  if (offset > 0 ) {
+    document.querySelector('#previous_button').disabled = false;
+  }
+
+  // If Meilisearch returns 20 items or less, 
+  // we are on the last page
+  if (results.hits.length < 21 ) {
+    document.querySelector('#next_button').disabled = true;
+  } 
+
+  // If Meilisearch returns exactly 21 results
+  // and our page can only show 20 items at a time,
+  // we have at least one more page with one result in it
+  if (results.hits.length === 21 ) {
+    document.querySelector('#next_button').disabled = false;
+  }
+}
+
+document.querySelector('#previous_button').onclick = function () { updatePageNumber(this) }
+document.querySelector('#next_button').onclick = function () { updatePageNumber(this) }
+```
+
+## Not recommended: Page selection
+
+This type of pagination consists of a numbered list of pages accompanied by next and previous buttons.
+
+![placeholder page selection UI](https://vuejsexamples.com/content/images/2018/11/vue-pagination.gif)
```

Comment:
Do we want to keep that one @guimachiavelli? (after our discussion yesterday, it may be outdated, I'm just wondering!)